projects/plugins/jetpack/extensions/blocks/ai-assistant/use-suggestions-from-openai.js

```diff
@@ -218,12 +220,35 @@ const useSuggestionsFromOpenAI = ( {
 			const data = JSON.parse( e.data );
 			const chunk = data.choices[ 0 ].delta.content;
 			if ( chunk ) {
-				fullMessage += chunk;
-				setAttributes( {
-					content: fullMessage,
-				} );
-				debug( fullMessage );
-				// debug( chunk );
+				// Check if the first chunks form the error marker by buffering them until we know for sure
+				if ( ! fullMessage ) {
+					initialBuffer += chunk;
```

Comment:
First feedback is `initialBuffer` can be defined in this scope if I'm not wrong. 

```suggestion
					let initialBuffer += chunk;
```

Reply:
The second comment is `initialBuffer` enough to check the string in the content?

Also, I have to admit that the double-checking is confusing, but I think what you want to check is when the message starts, or partially starts, with the `__JETPACK_AI_ERROR__` string, for instance:

```
__JETPACK_AI_ERROR__ ...
```

```
__JETPACK_AI_ER
```

is that correct?

Reply:
Yes. The idea behind the buffer is to avoid changing the content to, say `__JETPACK_AI_ER` before finally getting the full error. That is what the `'__JETPACK_AI_ERROR__'.startsWith( initialBuffer )` checks for.

The inverse check, `initialBuffer.startsWith( '__JETPACK_AI_ERROR__' )` is because the received tokens can make the content go from `__JETPACK_AI_ERRO` to `__JETPACK_AI_ERROR__: N`, for instance, in a single event. If that happens we know the error is there and we can kill the connection and show the error.

Reply:
Yes, `fullMessage` should be enough. Updating...

Reply:
This process also should be done on the server side.